run.go

```diff
@@ -0,0 +1,163 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/bitrise-io/go-steputils/command/gems"
+	"github.com/bitrise-io/go-steputils/v2/ruby"
+	"github.com/bitrise-io/go-utils/errorutil"
+	"github.com/bitrise-io/go-utils/fileutil"
+	"github.com/bitrise-io/go-utils/pathutil"
+	"github.com/bitrise-io/go-utils/v2/command"
+	"github.com/bitrise-io/go-xcode/appleauth"
+)
+
+// RunOpts ...
+type RunOpts struct {
+	WorkDir         string
+	AuthCredentials appleauth.Credentials
+	LaneOptions     []string
+	UseBundler      bool
+	GemVersions     gemVersions
+	EnableCache     bool
+}
+
+// Run ...
+func (f FastlaneRunner) Run(opts RunOpts) error {
+	// Run fastlane
+	f.logger.Println()
+	f.logger.Infof("Run Fastlane")
+
+	var envs []string
+	authEnvs, err := FastlaneAuthParams(opts.AuthCredentials)
+	if err != nil {
+		return fmt.Errorf("Failed to set up Fastlane authentication parameters: %v", err)
+	}
+	var globallySetAuthEnvs []string
+	for envKey, envValue := range authEnvs {
+		if _, set := os.LookupEnv(envKey); set {
+			globallySetAuthEnvs = append(globallySetAuthEnvs, envKey)
+		}
+
+		envs = append(envs, fmt.Sprintf("%s=%s", envKey, envValue))
+	}
+	if len(globallySetAuthEnvs) != 0 {
+		f.logger.Warnf("Fastlane authentication-related environment varibale(s) (%s) are set, overriding.", globallySetAuthEnvs)
+		f.logger.Infof("To stop overriding authentication-related environment variables, please set Bitrise Apple Developer Connection input to 'off' and leave authentication-related inputs empty.")
+	}
+
+	buildlogPth := ""
+	if tempDir, err := pathutil.NormalizedOSTempDirPath("fastlane_logs"); err != nil {
+		f.logger.Errorf("Failed to create temp dir for fastlane logs, error: %s", err)
```

Comment:
Pls use the warning log level if an error is not fatal.

Reply:
63dbf2112ebc7f41da68a21304c976f642ca1a03